flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/sink/writer/MongoWriter.java

```diff
@@ -105,10 +115,35 @@ public MongoWriter(
 
         // Initialize the mongo client.
         this.mongoClient = MongoClients.create(connectionOptions.getUri());
+
+        if (!flushOnlyOnCheckpoint() && writeOptions.getBatchIntervalMs() > 0) {
+            this.scheduler =
+                    Executors.newScheduledThreadPool(1, new ExecutorThreadFactory("mongo-writer"));
+
+            this.scheduledFuture =
+                    this.scheduler.scheduleWithFixedDelay(
+                            () -> {
+                                synchronized (MongoWriter.this) {
+                                    if (!closed && isOverMaxBatchIntervalLimit()) {
+                                        try {
+                                            doBulkWrite();
+                                        } catch (Exception e) {
+                                            flushException = e;
+                                        }
+                                    }
+                                }
+                            },
+                            writeOptions.getBatchIntervalMs(),
+                            writeOptions.getBatchIntervalMs(),
```

Comment:
The `batchIntervalMs` could be initialized earlier during construct the MongoDBWriter